app/src/ui/lib/ref-name-text-box.tsx

```diff
@@ -149,22 +144,38 @@ export class RefNameTextBox extends React.Component<
       return null
     }
 
-    const renderWarningMessage =
-      this.props.renderWarningMessage ?? this.defaultRenderWarningMessage
-
     return (
-      <div className="warning-helper-text">
-        <Octicon symbol={OcticonSymbol.alert} />
-
-        <p>{renderWarningMessage(sanitizedValue, proposedValue)}</p>
-      </div>
+      <InputWarning
+        id="branch-name-warning"
+        className="warning-helper-text"
+        trackedUserInput={proposedValue}
+        ariaLiveMessage={this.getWarningMessageAsString(
+          sanitizedValue,
+          proposedValue
+        )}
+      >
+        <p>{this.renderWarningMessage(sanitizedValue, proposedValue)}</p>
+      </InputWarning>
     )
   }
 
-  private defaultRenderWarningMessage(
+  private getWarningMessageAsString(
     sanitizedValue: string,
     proposedValue: string
-  ) {
+  ): string {
+    // If the proposed value ends up being sanitized as
+    // an empty string we show a message saying that the
+    // proposed value is invalid.
+    if (sanitizedValue.length === 0) {
+      return `Warning: ${proposedValue} is not a valid name.`
+    }
+
+    return `Warning: Will be ${
+      this.props.warningMessageVerb ?? 'saved '
+    } as ${sanitizedValue} (in kebab case).`
```

Comment:
I added "In kebab case" because the screen reader reads "this is a default branch" the same as "this-is-a-default-branch".

Reply:
Tho, thinking about, this kebab call out might not be bad addition the visual one just to clarify the change being made. Other thoughts, we could also say will be "(spaces replaced by hyphens)",  because we do have user who are not programmers, but kebab case is easy term to search.

Reply:
I'm not sure if `kebab` is very inclusive

Reply:
Thanks for bringing that to my attention. I was not aware of the negative use case of the word till googling after your comment. I have opened a[ PR to use the descriptive text instead.](https://github.com/desktop/desktop/pull/17866)